src/snakemake/resources.py

```
@@ -1,160 +1,65 @@
-from collections import UserDict, defaultdict
+from __future__ import annotations
+from collections import defaultdict
+import copy
 from dataclasses import dataclass
```

Ruff (0.11.9)

`dataclasses.dataclass` imported but unused

Remove unused import: `dataclasses.dataclass`

(F401)